menas/ui/components/dataset/datasetDetail.controller.js

```diff
@@ -370,15 +370,19 @@ sap.ui.define([
 
       //transform properties into a bindable format
       const aPropTemplates = this._model.getProperty(`/properties`) || [];
-      if(currentDataset && currentDataset.properties) {
+      if(currentDataset) {
         currentDataset.propertiesDisplay = [];
-        for(let sName in currentDataset.properties) {
-          const oPropTemplMatch = aPropTemplates.find((oTempl) => { return oTempl.name === sName });
-          const iOrder = oPropTemplMatch ? oPropTemplMatch.order : Infinity;
+
+        for(let property of aPropTemplates) {
+          let propValue = 'Missing';
+          if(currentDataset.properties && currentDataset.properties[property.name]) {
+            propValue = currentDataset.properties[property.name];
+          }
           currentDataset.propertiesDisplay.push({
-            name: sName,
-            value: currentDataset.properties[sName],
-            order: iOrder
+            name: property.name,
+            value: propValue,
+            order: property.order,
+            essentiality: property.essentiality._t
```

Comment:
I would maybe personally prefer a more functional approach, for Javascript, it's fine, I guess. It seems correct.